### PR TITLE
Test with more Ruby versions

### PR DESCRIPTION
**Pull request by betesh-at-data-axle**

### CODE REVIEW

**Comment by betesh-at-data-axle** [comment]
@matthuhiggins FYI

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        ruby: [ '2.4.10', '2.5.9', '2.6.9', '2.7.5', '3.0.3', '3.1.0' ]
```
Comment (matthuhiggins) [inline .github/workflows/ruby.yml:12]: What versions do you actually use this with? I'd prefer not to support ruby before 2.7

File: .simplecov [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  enable_coverage :branch
+  add_group "Aggregations", "lib/arelastic/aggregations"
+  add_group "Arities", "lib/arelastic/arities"
+  add_group "Builders", "lib/arelastic/builders"
+  add_group "Mappings", "lib/arelastic/mappings"
+  add_group "Nodes", "lib/arelastic/nodes"
+  add_group "Queries", "lib/arelastic/queries"
+  add_group "Searches", "lib/arelastic/searches"
+  add_group "Sorts", "lib/arelastic/sorts"
+end
```
Comment (matthuhiggins) [inline .simplecov:11]: The groupings seem unnecessary.

**Review by matthuhiggins — COMMENTED** [review]
Thanks for moving this to github workflows!

I'd like to reduce some of the extra changes here. See comments.

File: .simplecov (hunk 2 above)
Comment (betesh-at-data-axle) [reply .simplecov:11]: I found it helpful but it's your call.  I'll remove it if that's what you want.

Comment (matthuhiggins) [reply .simplecov:11]: Ok, leave this.

File: .simplecov [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+SimpleCov.minimum_coverage line: 97.8, branch: 97.2
+SimpleCov.minimum_coverage_by_file line: 70, branch: 92.8
```
Comment (matthuhiggins) [inline .simplecov:14]: Do these rules break the build if violated, or just change the report format?

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (betesh-at-data-axle) [reply .github/workflows/ruby.yml:12]: We can drop support for < 2.7.

File: .simplecov (hunk 3 above)
Comment (betesh-at-data-axle) [reply .simplecov:14]: They break it.  My intent is to ensure that contributors can't commit new code without writing tests for it.

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (betesh-at-data-axle) [reply .github/workflows/ruby.yml]: Done.

**Pull request merged**